smdebug/tensorflow/base_hook.py

```diff
@@ -482,19 +484,21 @@ def set_optimizer_variables(self, optimizer_variables):
         This method helps find the optimizer variables (such as momentum)
         :param optimizer_variables: list of tf.Variables/tf.Tensors/tf.MirroredVariables
         """
-        # TF 2.x doesn't provide gradient/optimizer variable names and values by default.
-        # Skipping set_gradients and set_optimizer_variables for Tf 2.x until there is
+        # TF 2.x provides only symbolic gradient variables that do not provide access to their values.
```

Comment:
you can remove this comment from set_optimizer_variables

Reply:
Done;

Reply:
I have also moved _write_optimizers to save_post_step